app/queries/arel_helpers.rb

```diff
@@ -0,0 +1,34 @@
+module Queries
+  module ArelHelpers
+    CTE = Struct.new(:table, :composed_cte)
+
+    def table(table)
+      Arel::Table.new(table, ActiveRecord::Base)
+    end
+
+    def cte(scope, as:)
```

Comment:
Going to attempt to change the API for this so it's more obvious. I don't like it as it is.